test/extended/router/gatewayapicontroller.go

```diff
@@ -673,3 +760,164 @@ func isOKD(oc *exutil.CLI) (bool, error) {
+			return false, nil
+		}
+
+		readyReplicas := deployment.Status.ReadyReplicas
+		e2e.Logf("The ready replicas is %v", readyReplicas)
+		if readyReplicas != 1 {
+			e2e.Logf(`The deployment %s in %s namespace is not ready(AvailableReplicas: %v), retrying`, deploymentName, ns, deployment.Status.AvailableReplicas)
+			return false, nil
+		}
+		return true, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to wait the istio is created successfully by checking its readyReplicas
+func pollWaitIstioCreated(oc *exutil.CLI, ingressNamespace, istioName string) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		readyReplicas, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ingressNamespace, "istio/"+istioName, `-o=jsonpath={.status.revisions.ready}`).Output()
+		if err != nil {
+			e2e.Logf("Failed to check istio %s, error: %v, retrying...", istioName, err)
+			return false, nil
+		}
+
+		if readyReplicas != "1" {
+			e2e.Logf("Wait for the ready replicas to be 1, and got %s", readyReplicas)
+			return false, nil
+		}
+		return true, nil
+
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to wait for the gateway lb service is automatically recreated successfully
+func pollWaitGWLBServiceRecreated(oc *exutil.CLI, ingressNamespace, gatewayLbService string, createdTime1 metav1.Time) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		lbService, err := oc.AdminKubeClient().CoreV1().Services(ingressNamespace).Get(context.Background(), gatewayLbService, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			e2e.Logf("Error getting the gateway lb service %s: %v, retrying...", gatewayLbService, err)
+			return false, nil
+		}
+
+		createdTime2 := lbService.ObjectMeta.CreationTimestamp
+		if createdTime2 == createdTime1 {
+			return false, nil
+		}
+
+		lb := lbService.Status.LoadBalancer
+		searchInfo := regexp.MustCompile("(IP:([0-9\\.a-fA-F:]+))|(Hostname:([0-9\\.\\-a-zA-Z]+))").FindStringSubmatch(lb.String())
+		if len(searchInfo) > 0 {
+			if gwlb := searchInfo[2]; len(gwlb) > 0 {
+				e2e.Logf("New load balancer ip %s is available", gwlb)
+				return true, nil
+			}
+			if gwlb := searchInfo[4]; len(gwlb) > 0 {
+				e2e.Logf("New load balancer hostname %s is available", gwlb)
+				return true, nil
+			}
+		}
+		e2e.Logf("Failed to get the new IP or hostname of the gateway lb service %s, retrying...", gatewayLbService)
+		return false, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to wait for the gateway dnsrecords is automatically recreated successfully
+func pollWaitGWDnsrecordsRecreated(oc *exutil.CLI, ingressNamespace, targetsList1 string) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		dnsrecordList, err := oc.AdminIngressClient().IngressV1().DNSRecords(ingressNamespace).List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
Let's be consistent with the other polling functions: if list fails we should log it and retry.

Reply:
Sure, I will update it.